cabot/cabotapp/models.py

```diff
@@ -489,28 +489,35 @@ def should_run(self):
         next_run_time = self.last_run + timedelta(minutes=self.frequency)
         return timezone.now() > next_run_time
 
+    @transaction.atomic()
     def run(self):
         start = timezone.now()
         try:
-            result = self._run()
-        except SoftTimeLimitExceeded as e:
-            result = StatusCheckResult(check=self)
-            result.error = u'Error in performing check: Celery soft time limit exceeded'
-            result.succeeded = False
+            result, tags = self._run()
+        except SoftTimeLimitExceeded:
+            result = StatusCheckResult(check=self, succeeded=False,
+                                       error=u'Error in performing check: Celery soft time limit exceeded')
+            tags = ['celery_timeout']
         except Exception as e:
-            result = StatusCheckResult(check=self)
-            result.error = u'Error in performing check: %s' % (e,)
-            result.succeeded = False
+            result = StatusCheckResult(check=self, succeeded=False, error=u'Error in performing check: %s' % (e,))
+            tags = ['run_error']
```

Comment:
I really like this...this opens up the possibility fo sending infra issues (like the celery timeout) to a different oncall for infrastructure so we don't wake everyone up when there's an infra-level issue